### PR TITLE
fix: mitigate ReDoS vulnerability with timeout + pattern validation

### PR DESCRIPTION
**Pull request by aleks-tpom6oh**
## Summary

- Add defense-in-depth ReDoS protection for regex pattern matching in AvoEventValidator
- **Pattern validator**: `isPatternPotentiallyDangerous:` rejects patterns with nested quantifiers (e.g. `(a+)+$`) before compilation
- **Timeout wrapper**: `safeNumberOfMatchesWithRegex:inString:timeout:` uses GCD semaphore with 2-second timeout as safety net
- Fail-open: timed-out or rejected patterns skip the constraint (no false positives)

## Why

`NSRegularExpression` uses ICU's backtracking engine. A malicious or compromised pattern like `(a+)+$` matched against a long string causes exponential CPU usage (ReDoS). No good RE2 library exists for Objective-C without heavy C++ dependencies, so we use a timeout + pattern validation approach instead.

## Changes

| File | Change |
|------|--------|
| `AvoEventValidator.h` | Declare new protection methods |
| `AvoEventValidator.m` | Add `isPatternPotentiallyDangerous:`, `safeNumberOfMatchesWithRegex:inString:timeout:`, update callers |
| `AvoEventValidatorTests.m` | Add 4 tests: dangerous pattern rejection, safe pattern allowance, timeout safety, end-to-end skip |

## Test plan

- [x] Dangerous patterns `(a+)+$`, `(.*)*`, `([a-z]+)*` rejected by pre-check
- [x] Safe patterns `^[0-9]+$`, `^[a-zA-Z]{1,100}$` pass pre-check
- [x] Timeout returns `NSNotFound` within 5 seconds for evil input
- [x] End-to-end: dangerous regex constraint silently skipped during validation
- [ ] Existing test suite passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Implemented timeout protection for regex operations to prevent application hangs
  * Added automatic detection and rejection of potentially dangerous regex patterns
  * Improved error handling and logging for regex-related operations

* **Tests**
  * Added comprehensive test coverage for regex safety features and edge cases

### CODE REVIEW

File: AvoInspector/Classes/AvoEventValidator.m [hunk 1]
```diff
@@ -533,6 +538,61 @@ + (void)checkMinMaxRanges:(NSDictionary<NSString *, NSArray<NSString *> *> *)min
...
+    if (timedOut != 0) {
+        if ([AvoInspector isLogging]) {
+            NSLog(@"[avo] Avo Inspector: Regex execution timed out after %.0fs for pattern '%@'",
+                  timeout, regex.pattern);
+        }
+        return NSNotFound;
+    }
+
+    return result;
+}
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEventValidator.m:598]: fixed
